server/base/src/main/java/org/apache/accumulo/server/AbstractServer.java

```diff
@@ -87,6 +94,16 @@ public void runServer() throws Exception {
     }
   }
 
+  @Override
+  public void registerMetrics(MeterRegistry registry) {
+    processMetrics.registerMetrics(registry);
+  }
+
+  public void initServerMetrics(MetricsProducer... producers) {
```

Comment:
I don't think this method is necessary, and it's initializing the metrics for AbstractServer differently than everything else. Since you have modified `AbstractServer` to implement `MetricsProducer`, then:

  1. You should have any subclasses of `AbstractServer` call `super.registerMetrics` in their respective `registerMetrics` method. For example, [Compactor.registerMetrics](https://github.com/apache/accumulo/blob/15c9f55280f4ce28a80b800f1e05128f33c85db3/server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java#L170).
  2. You should pass `this` in subclasses of `AbstractServer` in the call to `MetricsUtil.initializeProducers`.

Reply:
@dlmarion What do you think is easier to understand?  Classes that extended abstract server must call `super.registerMetrics` or call a method 'initServerMetrics' ?  Either way requires something that may not be obvious.  

Because of the tagging, the registration / initialization needs to occur in the run method - and after certain things have been initialized (host and port) 

I'm not thrilled with either approach, but went with a specific `initServerMetrics` method - if relying on calling `super.registerMetrics' and the needing to call the static `MetricsUtil.initializeProducers` which results in an indirect call to `registerMetrics`?

I was thinking about moving the static `initializeMetrics` out of MetricsUtil and put the functionality into AbstractServer - I think that functionality only applies to services extending AbstractServer.  

The `initializeProducers` functionality needs to exist to register metrics producers for things that are created outside of the AbstarctServer context (thrift metrics, cache,....) - but those are created / exist within a service context that has already been expected to initialize the registry.

Reply:
Addressed in 49b93b349e - backed of removal of MetricsUtil for now to minimize changes.  May look into additional changes in future PRs, but this code works as expected.